src/main/java/org/apache/maven/buildcache/checksum/MavenProjectInput.java

```diff
@@ -684,51 +804,58 @@ private SortedMap<String, String> getMutableDependenciesHashes(String keyPrefix,
                         projectInputCalculator.calculateInput(dependencyProject).getChecksum();
             } else // this is a snapshot dependency
             {
-                DigestItem resolved = resolveArtifact(repoSystem.createDependencyArtifact(dependency), false);
+                DigestItem resolved = null;
+                try {
+                    resolved = resolveArtifact(dependency);
+                } catch (ArtifactResolutionException | InvalidVersionSpecificationException e) {
+                    throw new IOException(e);
+                }
                 projectHash = resolved.getHash();
             }
             result.put(
-                    keyPrefix + KeyUtils.getVersionlessArtifactKey(repoSystem.createDependencyArtifact(dependency)),
-                    projectHash);
+                    keyPrefix + KeyUtils.getVersionlessArtifactKey(createDependencyArtifact(dependency)), projectHash);
         }
         return result;
     }
 
     @Nonnull
-    private DigestItem resolveArtifact(final Artifact dependencyArtifact, boolean isOffline) throws IOException {
-        ArtifactResolutionRequest request = new ArtifactResolutionRequest()
-                .setArtifact(dependencyArtifact)
-                .setResolveRoot(true)
-                .setResolveTransitively(false)
-                .setLocalRepository(session.getLocalRepository())
-                .setRemoteRepositories(project.getRemoteArtifactRepositories())
-                .setOffline(session.isOffline() || isOffline)
-                .setForceUpdate(session.getRequest().isUpdateSnapshots())
-                .setServers(session.getRequest().getServers())
-                .setMirrors(session.getRequest().getMirrors())
-                .setProxies(session.getRequest().getProxies());
-
-        final ArtifactResolutionResult result = repoSystem.resolve(request);
-
-        if (!result.isSuccess()) {
+    private DigestItem resolveArtifact(final Dependency dependency)
+            throws IOException, ArtifactResolutionException, InvalidVersionSpecificationException {
+
+        org.eclipse.aether.artifact.Artifact dependencyArtifact = new org.eclipse.aether.artifact.DefaultArtifact(
+                dependency.getGroupId(),
+                dependency.getArtifactId(),
+                dependency.getClassifier(),
+                null,
+                dependency.getVersion(),
+                new DefaultArtifactType(dependency.getType()));
+        ArtifactRequest artifactRequest = new ArtifactRequest().setArtifact(dependencyArtifact);
+
+        ArtifactResult result = artifactResolver.resolveArtifact(session.getRepositorySession(), artifactRequest);
```

Comment:
-1, this is `impl` package, use public API instead instead:
https://github.com/apache/maven-resolver/blob/master/maven-resolver-api/src/main/java/org/eclipse/aether/RepositorySystem.java#L164-L178

Reply:
fixed